src/app/wallets/update-pending-payments.ts

```diff
@@ -0,0 +1,187 @@
+  if (lndService instanceof Error) return lndService
+
+  const { paymentHash, pubkey } = paymentLiabilityTx
+  // If we had PaymentLedgerType => no need for checking the fields
+  if (!paymentHash)
+    return new InconsistentDataError("paymentHash missing from payment transaction")
+  if (!pubkey) return new InconsistentDataError("pubkey missing from payment transaction")
+
+  const lnPaymentLookup = await lndService.lookupPayment({
+    pubkey,
+    paymentHash,
+  })
+  if (lnPaymentLookup instanceof Error) {
+    const lightningLogger = logger.child({
+      topic: "payment",
+      protocol: "lightning",
+      transactionType: "payment",
+      onUs: false,
+    })
+    lightningLogger.error({ err: lnPaymentLookup }, "issue fetching payment")
+    return lnPaymentLookup
+  }
+  const { status, roundedUpFee } = lnPaymentLookup
+
+  if (status === PaymentStatus.Settled || status === PaymentStatus.Failed) {
+    const ledgerService = LedgerService()
+    const settled = await ledgerService.settlePendingLiabilityTransactions(paymentHash)
+    if (settled instanceof Error) {
+      paymentLogger.error({ error: settled }, "we didn't have any transaction to update")
+      return settled
+    }
+
+    // PaymentStatus.Failed transactions have null 'roundedUpFee'
+    if (status === PaymentStatus.Settled && roundedUpFee !== null) {
+      paymentLogger.info(
+        { success: true, id: paymentHash, payment: paymentLiabilityTx },
+        "payment has been confirmed",
+      )
+      return reimburseFee({
+        liabilitiesAccountId,
+        paymentLiabilityTx,
+        roundedUpFee,
+        logger,
+      })
+    }
+    return revertTransaction({
+      paymentLiabilityTx,
+      lnPaymentLookup,
+      logger: paymentLogger,
+    })
+  }
+}
+
+const reimburseFee = async ({
+  liabilitiesAccountId,
+  paymentLiabilityTx,
+  roundedUpFee,
+  logger,
+}: {
+  liabilitiesAccountId: LiabilitiesAccountId
+  paymentLiabilityTx: LedgerTransaction
+  roundedUpFee: Satoshis
+  logger: Logger
+}): Promise<void | ApplicationError> => {
+  const { paymentHash } = paymentLiabilityTx
+  // If we had PaymentLedgerType => no need for checking the fields
+  if (!paymentHash)
+    return new InconsistentDataError("paymentHash missing from payment transaction")
+
+  if (!paymentLiabilityTx.feeKnownInAdvance) {
+    const feeDifference = FeeReimbursement(paymentLiabilityTx.fee).getReimbursement({
+      actualFee: roundedUpFee,
+    })
+    if (feeDifference === null) {
+      logger.warn(
+        `Invalid reimbursement fee for ${{
+          maxFee: paymentLiabilityTx.fee,
+          actualFee: roundedUpFee,
+        }}`,
+      )
+      return
+    }
+
+    logger.info(
+      {
+        paymentResult: paymentLiabilityTx,
+        feeDifference,
+        maxFee: paymentLiabilityTx.fee,
+        actualFee: roundedUpFee,
+        id: paymentHash,
+      },
+      "logging a fee difference",
+    )
+
+    const price = await PriceService().getCurrentPrice()
```

Comment:
In cases where `updatePendingPayments` is locked and called together with `updatePendingInvoices`, can this price value retrieved diverge within the two implementations, and if yes is that ok?

Our existing implementation has logic for caching the price I believe (`setCurrentPrice` method)

Reply:
I don't see this as an issue. In the existing implementation `setCurrentPrice` gets called earlier in the request/response cycle but I cannot see how this should have an effect on behaviour.